packages/openneuro-app/src/scripts/errors/errorBoundary.jsx

```diff
@@ -61,46 +142,32 @@ class ErrorBoundary extends React.Component {
   }
 
   render() {
-    const { supportModal } = this.state
+    const { supportModal, message } = this.state
     const { subject, description } = this.props
     const error = this.props.error || this.state.error
-    return this.state.hasError ? (
-      <>
-        <p className="generic-error-message">
-          {this.props.errorMessage || 'An error has occurred.'}
-          <br />
-          Please support us by documenting the issue with{' '}
-          <a onClick={this.openSupportModalFromLink}>
-            <u>FreshDesk</u>
-          </a>
-          .
-        </p>
-        {supportModal && (
-          <Overlay>
-            <ModalContainer>
-              <ExitButton onClick={this.closeSupportModal}>&times;</ExitButton>
-              <h3>Support</h3>
-              <hr />
-              <div>
-                To ensure that we can quickly help resolve this issue, please
-                provide as much detail as you can, including what you were
-                trying to accomplish when the error occurred.
-              </div>
-              <FreshdeskWidget
-                {...{
-                  subject,
-                  description,
-                  error,
-                  sentryId: this.state.eventId,
-                }}
-              />
-            </ModalContainer>
-          </Overlay>
-        )}
-      </>
-    ) : (
-      this.props.children
-    )
+    const { loading } = this.props
+
+    if (error) {
+      if (message) {
+        return <DatasetRedirect message={message} />
+      } else if (loading) {
```

Comment:
We can remove this conditional. The loading state precedes any response, so if we wanted to include a spinner for it that would be the first condition to handle.

Reply:
Makes sense. On second gander, I also think the final else (return children) can simply be a return statement given the 'blanket' error check.